pkg/resources/aws/tags.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/golang/glog"
+	"k8s.io/kops/upup/pkg/fi/cloudup/awsup"
+)
+
+// HasOwnedTag looks for the new tag indicating that the cluster does owns the resource, or the legacy tag
+func HasOwnedTag(description string, tags []*ec2.Tag, clusterName string) bool {
```

Comment:
New function that checks for the owned tag, rather than checking for the shared tag.  i.e. assume _not_ owned, so we fail-safe.